src/app/pages/reports-dashboard/reports.service.ts

```diff
@@ -145,22 +145,23 @@ export class ReportsService implements OnDestroy {
     return data;
   }
 
-  getReportTabs(): ReportTab[] {
-    return Array.from(reportTypeLabels)
-      .filter(([value]) => {
-        if (value === ReportType.Ups && !this.hasUps) {
-          return false;
-        }
-
-        if (value === ReportType.Target && !this.hasTarget) {
-          return false;
-        }
-
-        return true;
-      })
-      .map(([value, label]) => {
-        return { value, label } as ReportTab;
-      });
+  getReportTabs(): Observable<ReportTab[]> {
+    return of(Array.from(reportTypeLabels)).pipe(
+      withLatestFrom(this.hasUps$, this.hasTarget$, this.hasDiskTemperature$),
```

Comment:
What was the advantage of making `getReportTabs` return Observable? It's still only going to emit once. Looks like you could do:
```
combineLatest([
       this.reportsService.getReportGraphs(),
       of(this.reportsService.getReportTabs()),
])
```
above.

Reply:
It seems overkill having tabs as an observable. We could rely on the `getReportGraphs` only.